03/app.js

```diff
@@ -3,19 +3,66 @@ import ReactDOM from 'react-dom';
 
 import Category from './Category';
 import Cart from './Cart';
+import Product from './Product';
 
 import data from './data.json';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
-    
+
+    addItem = (product) => {
+        product.isCart = true;
+
+        this.setState({
+            cart: [...this.state.cart, product]
+        });
+    }
+
+    removeItem = (product) => {
+        const copyItemsCart = [...this.state.cart];
+        product.isCart = false;
+        
+        const filterItems = copyItemsCart.filter(item => {
+            return item.id !== product.id;
+        });
+
+        this.setState({
+            cart: filterItems
+        });
+    }
+
+
+
     render() {
+        const productList = data.map(item => {
+            return(
+                <Product 
+                    prod = {item}
+                    name={item.name}
+                    price = {item.price}
+                    isCategory = {true}
+                    add = {this.addItem}
```

Comment:
zrobiłbym jeszcze jeden parametr `inCart={ this.state.cart.some(cartItem => cartItem.id  === item.id }`.
Raczej nie staramy się zmieniać danych, które mamy z zewnątrze - mówie o `data`.
Jeśli chciałbyś utrzymać swoją koncepcję to `data` wrzuciłbym do `state` i dam go modyfikował.
Ttedy miałbyś `state` z `cart` i np. `data`.